packages/calypso-e2e/src/lib/components/editor-site-styles-component.ts

```diff
@@ -128,6 +133,22 @@ export class EditorSiteStylesComponent {
 		await this.editorDimensionsComponent.resetAll();
 	}
 
+	/**
+	 * Sets a style variation for the site.
+	 * This auto-handles returning to top menu and navigating down.
+	 *
+	 * @param {StyleVariation} styleVariation The style variation to set.
+	 */
+	async setStyleVariation( styleVariation: StyleVariation ): Promise< void > {
+		await this.returnToTopMenu();
+		await this.clickMenuButton( 'Browse styles' );
+		const locator =
+			styleVariation === 'Default'
+				? this.editor.locator( selectors.defaultStyleVariation )
+				: this.editor.locator( selectors.nonDefaultStyleVariation ).first();
```

Comment:
**comment:non-blocking** - the method name does not accurately portray the actual behaviour - I had expected this to take a parameter `name` that corresponds to the name shown in the sidebar.

Reply:
Unfortunately, the names of the style variations since they depend on the theme (e.g. the Twenty Twenty-Three theme has Default, Aubergine, Block out, ...; and the Zoologist theme has Default, Black, White, ...), and since we cannot guarantee the site will always have the same theme, we shouldn't hardcode those names (or they'll fail as soon as the site switches to a different theme).

The only thing we know is that all themes has a Default style variation, so we cannot really do anything more than picking the default or any non-default style variation.

Reply:
> unless the change is expected

I think this is key part here. For instance, the site used by this PR had its theme recently changed to Twenty Twenty-One which is completely breaking this E2E test since that theme doesn't support the site editor.

That theme switch was a manual action and I don't know if that's what'd we considered an expected change that should affect the E2E tests.

Anyway, do you think we can change the theme back to a block theme such as Twenty Twenty-Three or Zoologist? If the site necessarily needs a legacy theme, we might need to create a a new site.

Reply:
> do you think we can change the theme back to a block theme such as Twenty Twenty-Three or Zoologist? 

I discussed this privately with @worldomonation, and we cannot activate a block theme just yet until https://github.com/Automattic/wp-calypso/pull/72300 is merged.

So I'm going to put this PR on hold for now since we cannot target any site where we can test Limited Global Styles.